### ISSUE TITLE
[Bug]: Desk

### ISSUE THREAD

**Issue by nicolai-smeup** [simplex-chat/simplex-chat#3425]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Platform

Windows, Android

### OS version

Android 13, Windows 11

### App version

v5.4.0-beta.4

### Current Behavior

It seems not to be possible to connect my account on Android with my PC. Scanning the QR code seems not to go through on phone (it goes "connecting" and then resets).

### Expected Behavior

After scanning the QR code the account should connect.

### Steps To Reproduce

- On phone use the function "Connect to desktop"
- On PC, "connect a phone"
- Scan the QR code with the phone

**Comment by epoberezkin** [comment]
Is the firewall configured to allow desktop app accept incoming network requests?

**Comment by nicolai-smeup** [comment]
Dear Evgeny,

I checked and it seems that the app is not blocked by Windows Firewall.

**Comment by Ciugam** [comment]
Hello,
I have the same issue, using Fedora, default firewall config (no custom blocking ports) and Android 14.
I do not open another issue, simply the procedure is the same as @nicolai-smeup and I got the same result, scan QR code from mobile does not do anything on desktop.

**Comment by N0tMy4cc0unt** [comment]
Same here: on windows deskop v5.4 build 18 core version v5.4.0.6; on android  v5.4 app build 162 core v5.4.0.6 - I had to install apk from github release since in the one from google playstore there were no option connect to desktop. I do not know how to install windows file from github release (the one without extension) to try that version.

**Comment by khuldraeseth** [comment]
I ran into the same. App version v5.4, build 18, core v5.4.0.6 from [AUR](https://aur.archlinux.org/packages/simplex-desktop-appimage) on desktop, app version v5.4, build 162, core v5.4.0.6 from [F-Droid](https://f-droid.org/packages/chat.simplex.app/) on mobile.

I've been able to connect with both devices on a different network. Won't be back on the original network for a little while to make sure, but I suspect it may have had some sort of client isolation.

**Comment by Rastem88** [comment]
I'm running into this problem. If I disable my machines local firewall, I can connect, but I am unable to once I re-enable. I'm running the 5.4.0 appimage on Arch and 5.4 from f-droid.

I've noticed on the connect screen on my desktop it says "Waiting to connect on port " ##### where the number changes. If I open this port I can connect. 

What ports should we enable to connect devices? I looked through what I could, but only saw documentation for opening tcp port 5223.

**Comment by space-d-n** [comment]
I faced the same issue with iphone and mac, for me it was because my phone was not connected to the same network as my laptop.

**Comment by nicolai-smeup** [comment]
Still trying but it doesn't work.... It is connected to the same Wi-Fi but seems not to be working...

**Comment by koba-ninkigumi** [comment]
@space-d-n 

> I faced the same issue with iphone and mac, for me it was because my phone was not connected to the same network as my laptop.

Is your problem happening when you read the QR code?
I can even read QR codes on iOS and Mac.
But when I press the "Confirm" button, the app crashes.
I have reported this issue.
https://github.com/simplex-chat/simplex-chat/issues/3488

I would like to know if the problem you are encountering is the same as mine or different.

**Comment by space-d-n** [comment]
> @space-d-n
> 
> > I faced the same issue with iphone and mac, for me it was because my phone was not connected to the same network as my laptop.
> 
> Is your problem happening when you read the QR code? I can even read QR codes on iOS and Mac. But when I press the "Confirm" button, the app crashes. I have reported this issue. #3488
> 
> I would like to know if the problem you are encountering is the same as mine or different.

No, my app didn't crash, as written in this issue it just "seems not to go through on phone", meaning that just nothing happened, until I switched my phone to the same network.

**Comment by mariaa144** [comment]
I couldn't connect to desktop from Android to NixOS either. I tried to turn off Mullvad on both desktop and Android but it still wouldn't connect. Can the desktop be connected to the phone over the public internet, not requiring a local LAN?

**Comment by deepfire** [comment]
@mariaa144, off topic -- I'm intrigued that you mention running SimplexChat on NixOS -- how are you doing this -- are you importing its repository through the flake?

**Comment by mariaa144** [comment]
No, I'm only using the AppImage and using `appimage-run` from `nixpkgs`.

**Comment by nikonico** [comment]
Did you find a solution? I'm also using NixOS and I'm having the same issue!

**Comment by dillfrescott** [comment]
Happening on Windows 11 for desktop OS and Android 14 for mobile OS. Sadly this might be a dealbreaker for me.

**Comment by caniko** [comment]
I also have this issue on NixOS :(

**Comment by eric72777** [comment]
To fix this issue on NixOS, open the tcp firewall port listed on the pairing screen. This number is different for each pairing qr code generated. For example,
```
networking.firewall.allowedTCPPorts = [ 37800 38989 ];
```

**Comment by caniko** [comment]
@eric72777 I didn't know I had a firewall 😆 

> ```nix
>networking.firewall.allowedTCPPorts = [ 37800 38989 ];
>```

You don't even need to set two ports one port is enough, just make sure to use that port in the application when you pair to desktop!

**Comment by Kreyren** [comment]
In case anyone gets confused by it like i did, the port it expects connection on is shown in:

![image](https://github.com/user-attachments/assets/04509cf1-6681-4659-b783-c75b76c23b7a)

The UI should probably be adjusted to make this more obvious instead of having it behind a hidden hint

**Comment by dillfrescott** [comment]
```
{
  "type": "connectionFailed",
  "chatError": {
    "type": "errorAgent",
    "agentError": {
      "type": "RCP",
      "rcpErr": {
        "type": "exception",
        "exception": "Network.Socket.connect: <socket: 174>: does not exist (No route to host)"
      }
    }
  }
}
```

I have no idea why it's doing this, I'll be honest. But I hope this is the right issue thread for the error.

**Comment by Drew-Daniels** [comment]
Thanks @eric72777, @caniko, and @Kreyren - updating the available TCP ports in my Nix config fixed the issue for me too.

**Comment by P0huber** [comment]
I'm experiencing issues linking the SimpleX applications as well on Desktop App and Mobile device **in the same local network with VPN enabled ONLY**. 

Desktop:
OS: MacOS 15.6.1 (aarch64)
App: SimpleX 6.4.4

Mobile:
OS: Android 14
App: SimpleX 6.4.4

Error:
`{
    "type": "connectionFailed",
    "chatError": {
        "type": "errorRemoteCtrl",
        "remoteCtrlError": {
            "type": "timeout"
        }
    }
}`

**Comment by CEKPETHO** [comment]
@epoberezkin when will fix this broken action?

**Comment by CEKPETHO** [comment]
@epoberezkin if don't have enough data, maybe you could allow us to get the data for you? Except:

```
{
    "type": "connectionFailed",
    "chatError": {
        "type": "errorRemoteCtrl",
        "remoteCtrlError": {
            "type": "timeout"
        }
    }
}
```

**Comment by Narasimha-sc** [comment]
> [@epoberezkin](https://github.com/epoberezkin) when will fix this broken action?

Hello, if your mobile app does not connect to desktop app, please check these things:
1. Check that both devices are connected to the same networks (e.g., it won't work if mobile is connected to mobile Internet and desktop to WiFi).
2. If you use VPN on mobile, allow connections to local network in your VPN settings (or disable VPN). 
3. Allow SimpleX Chat on desktop to accept network connections in system firewall settings. You may choose a specific port desktop app is using to accept connections, by default it uses a random port every time.
4. Check that your wifi router allows connections between devices (e.g., it may have an option for "device isolation", or similar).
5. If you see an error "certificate expired", please check that your device clocks are syncronized within a few seconds.
6. If iOS app fails to connect and shows an error containing "no route", check that local network connections are allowed for the app in system settings.

Also see this post: https://simplex.chat/blog/20231125-simplex-chat-v5-4-link-mobile-desktop-quantum-resistant-better-groups.html#link-mobile-and-desktop-apps-via-secure-quantum-resistant-protocol

If none of the suggestions work for you, you can create a separate profile on each device and create a small group inviting both of your device profiles and your contact.

**Comment by CEKPETHO** [comment]
> Hello, if your mobile app does not connect to desktop app, please check these things:

@Narasimha-sc thank you for taking the time to look into this, as well as for providing such informative and helpful feedback.

Unfortunately, however, the problem has remained unresolved for several years now. Before publishing my post, I had already tried all of the solutions you suggested, but unfortunately, none of them helped.
Furthermore, during these attempts I was able to collect some additional information, which suggests that the issue is directly related to the messenger itself.

I would be happy to provide more technical information if there were a way to obtain specific error messages or other data that could help identify the source of the problem. Unfortunately, I currently lack the technical expertise required to investigate the source code myself.

That said, I would like to emphasize that I have already checked all of the options and configurations that were suggested, and at this point, everything seems to indicate that the issue lies within the messenger itself.

**Comment by Narasimha-sc** [comment]
I can't reproduce the issue, it's working fine here, can you explain why do you think it's problem within the messenger?

**Comment by u2ix** [comment]
Also had issues connecting from GrapheneOS to MacOS with a VPN (Mullvad) enabled. First I thought enabling *Local Network sharing* in the VPN App as mentioned above would solve the issue, but I still got this error while connecting to the desktop app:
```
{
    "type": "connectionFailed",
    "chatError": {
        "type": "errorAgent",
        "agentError": {
            "type": "RCP",
            "rcpErr": {
                "type": "exception",
                "exception": "Network.Socket.connect: <socket: 155>: permission denied (Permission denied)"
            }
        }
    }
}
```

While debugging the issue, I discovered that also accessing a web server on the desktop was blocked by the Phone.
In the end, the only thing what solved the issue for me on **GrapheneOS** with a VPN was to go to the Settings:
`Network & Internet > VPN > VPN Settings` and Disable *Block connections without VPN*.
